### PR TITLE
Support scala 2.12.10 and spark 2.4.4

### PR DESCRIPTION
**Pull request by vikatskhay**
Hello.

Related to https://github.com/tensorflow/ecosystem/pull/141#issuecomment-618852587

Not sure whether it'd be better to support multiple scala versions, perhaps introducing multiple profiles could be a good idea. As I understood from the comments https://github.com/tensorflow/ecosystem/pull/141 scala 2.11 could be left behind. Please let me know if I understood it incorrectly.

Tested with:
```
mvn clean install
```

### CODE REVIEW

File: spark/spark-tensorflow-connector/src/main/scala/org/tensorflow/spark/datasources/tfrecords/DefaultSource.scala [hunk 1]
```diff
@@ -167,7 +167,7 @@ class DefaultSource extends DataSourceRegister
   }
 }
 
-object DefaultSource {
+object DefaultSource extends scala.Serializable {
```
Comment (vikatskhay) [inline spark/spark-tensorflow-connector/src/main/scala/org/tensorflow/spark/datasources/tfrecords/DefaultSource.scala:170]: https://github.com/tensorflow/ecosystem/pull/141/files#r332816724

File: spark/spark-tensorflow-connector/src/main/scala/org/tensorflow/spark/datasources/tfrecords/DefaultSource.scala [hunk 2]
```diff
@@ -189,7 +189,7 @@ object DefaultSource {
     // Make the directory if it does not exist
     dir.mkdirs()
     // The path to the partition file.
-    val filePath = localPath + s"/part-" + String.format("%05d", new java.lang.Integer(index))
+    val filePath = localPath + s"/part-" + String.format("%05d", java.lang.Integer.valueOf(index))
```
Comment (vikatskhay) [inline spark/spark-tensorflow-connector/src/main/scala/org/tensorflow/spark/datasources/tfrecords/DefaultSource.scala:192]: Constructor of Integer is deprecated, with `-Xfatal-warnings` the build fails.

**Comment by vikatskhay** [comment]
@googlebot I signed it!

**Comment by jhseu** [comment]
@skavulya Mind taking a look?

**Comment by groodt** [comment]
I would really like a Scala 2.12 version to be published as well.

**Comment by skavulya** [comment]
@vikatskhay Thanks for the contribution. It looks good.
@jhseu The PR is ready for merge.

**Comment by vikatskhay** [comment]
Thanks heaps for reviewing this PR. What should be my next steps to get it merged?

**Comment by vikatskhay** [comment]
Quick question: when could the new version be released including this change?

**Comment by groodt** [comment]
@jhseu @skavulya Does there need to be a new release to maven central now that this has been merged?

**Comment by skavulya** [comment]
maven central gets updated whenever a new TensorFlow version is released

**Comment by groodt** [comment]
> maven central gets updated whenever a new TensorFlow version is released

Hmmm... is there a technical reason for this? It doesn't seem to be the case though, since there have been numerous TensorFlow releases since Aug 2018, but Maven central is showing that the last time this connector was released was in Aug 2018.

https://mvnrepository.com/artifact/org.tensorflow/spark-connector_2.11/1.10.0

**Comment by skavulya** [comment]
The connector is https://mvnrepository.com/artifact/org.tensorflow/spark-tensorflow-connector

There is no update since TensorFlow 2.0

@jhseu What would we need to do to update the release scripts?

**Comment by NitinAgg** [comment]
It seems this did not get updated with TF 2.2.

**Comment by vikatskhay** [comment]
Would be so great to get this out on maven central.

**Pull request merged**